web/crux/src/app/agent/agent.service.ts

```diff
@@ -364,8 +359,7 @@ export default class AgentService {
     connection: GrpcNodeConnection,
     request: AgentInfo,
   ): Promise<Observable<AgentCommand>> {
-    const updatedAgent = this.agents.has(request.id)
-    if (updatedAgent) {
+    if (this.agents.has(request.id)) {
       const agent = this.agents.get(request.id)
```

Comment:
You can just `.get()` the agent and check for` !agent`, so you don't need to lookup the key twice in the hashtable.